openie/src/main/scala/org/clulab/openie/entities/CustomizableRuleBasedFinder.scala

```diff
@@ -48,34 +52,33 @@ class CustomizableRuleBasedFinder(
     } else {
       // check that our expanded entities haven't swallowed any avoid mentions
       val avoidLabel = avoid.head.labels.last
-      trimmedEntities.filterNot { m => stateFromAvoid.hasMentionsFor(m.sentence, m.tokenInterval, avoidLabel) }
+      trimmedEntities.filterNot { m => _avoidState.hasMentionsFor(m.sentence, m.tokenInterval, avoidLabel) }
     }
     res
   }
 
   /**
-   * Determines whether or not an entity is a valid base entity. We want to disallow JJ-only entities except
-   * when they are a result of the head noun being a trigger (i.e. being avoided)
+   * Determines whether or not an entity is a valid base entity.
    */
   def isValidBaseEntity(entity: Mention): Boolean = {
-    containsValidNounVerb(entity)
-  }
-
-  /**
-   * Determines if the entity has at least one noun/verb that is not a stopNER
-   * @param entity the candidate entity Mention
-   */
-  def containsValidNounVerb(entity: Mention): Boolean = {
-    //val lemmas = entity.lemmas.get
     val tags = entity.tags.get
     val entities = entity.entities.get
 
     // Make sure there is a noun that isn't a specifically excluded  named entity.
     tags.indices.exists { i =>
-      (tagSet.isAnyNoun(tags(i)) || tagSet.isAnyVerb(tags(i))) && !stopNER.contains(entities(i))
+      isValidTag(tags(i)) && !stopNER.contains(entities(i))
     }
   }
 
+  /**
+   * Determines if the tag is a noun/verb
+   * @param tag the POS tag to consider
+   */
+  def isValidTag(tag: String): Boolean = {
+    if (ensureBaseTagNounVerb) tagSet.isAnyNoun(tag) || tagSet.isAnyVerb(tag) else true
```

Comment:
This might be written as
```scala
!ensureBaseTagNounVerb || tagSet.isAnyNoun(tag) || tagSet.isAnyVerb(tag)
```

Reply:
That specific line 78, I mean.

Reply:
yes, we could write it that way -- I thought this was a little more readable/parseable, but one man's conditional is another mans.... 
I can change if you want (else you can)